### PR TITLE
Migrate to use Ballerina gradle plugin and centralized workflows

### PR DESCRIPTION
**Pull request by NipunaRanasinghe**
# Description

Fixes https://github.com/ballerina-platform/ballerina-library/issues/5814

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

**Test Configuration**:
* Ballerina Version:
* Operating System:
* Java SDK: 

# Checklist:

### Security checks
 - [ ] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? 
 - [ ] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?

### CODE REVIEW

File: ballerina/build.gradle [hunk 1]
```diff
@@ -33,46 +40,68 @@ def stripBallerinaExtensionVersion(String extVersion) {
...
-        return extVersion.replace("${project.ext.snapshotVersion}", '')
+        return extVersion.replace("${project.ext.snapshotVersion}", "")
     }
 }
 
+ballerina {
+    packageOrganization = packageOrg
+    module = packageName
+    testCoverageParam = "--code-coverage --coverage-format=xml"
+    buildOnDockerImage = "nightly"
```
Comment (AzeemMuzammil) [inline ballerina/build.gradle:51]: Isn't it required to change to the relevant Ballerina build version? This is what will reflect on Central after a publish.

File: ballerina/Ballerina.toml [hunk 2]
```diff
@@ -9,84 +9,94 @@ icon = "icon.png"
...
 
-[[platform.java11.dependency]]
+[[platform.java17.dependency]]
 groupId = "io.netty"
 artifactId = "netty-transport-native-epoll"
 version = "4.1.94.Final"
+path = "./lib/netty-transport-native-epoll-4.1.94.Final.jar"
 
-[[platform.java11.dependency]]
+[[platform.java17.dependency]]
```
Comment (ThisaruGuruge) [inline ballerina/Ballerina.toml:86]: What is the conclusion for managing the transitive native dependencies in the repo itself?

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -1,72 +1,18 @@
 name: CI
```
[the remark this answers is not in the document]
Comment (ThisaruGuruge) [reply .github/workflows/ci.yml:1]: +1 to use "Build"

[the remark this answers is not in the document]
Comment (NipunaRanasinghe) [reply .github/workflows/ci.yml]: Changed with 90f9af4

File: .github/workflows/release.yml [hunk 4]
```diff
@@ -1,63 +1,16 @@
...
-      - name: Ballerina Push
-        run: bal push
-        working-directory: ./ballerina
-        env:
-          BALLERINA_CENTRAL_ACCESS_TOKEN: ${{ secrets.BALLERINA_CENTRAL_ACCESS_TOKEN }}
-          JAVA_HOME: /usr/lib/jvm/default-jvm
+  call_workflow:
+    name: Run Release Workflow
+    if: ${{ github.repository_owner == 'ballerina-platform' }}
+    uses: ballerina-platform/ballerina-standard-library/.github/workflows/release-package-template.yml@main
```
[the remark this answers is not in the document]
Comment (NipunaRanasinghe) [reply .github/workflows/release.yml]: Changed with 
90f9af4

File: ballerina/Ballerina.toml (hunk 2 above)
Comment (NipunaRanasinghe) [reply ballerina/Ballerina.toml:86]: Had a discussion with @Dilhasha and it seems that declaring the transitive native dependencies in the Ballerina.toml is the recommended options to minimize dependency conflicts.

File: ballerina/build.gradle (hunk 1 above)
Comment (NipunaRanasinghe) [reply ballerina/build.gradle:51]: As per the offline discussion I had with @ThisaruGuruge, we need to have some changes from our Ballerina gradle plugin side to address this. IMO we can keep this as is, until we finalize it.

**Review by ThisaruGuruge — APPROVED** [review]

**Pull request merged**